contracts/Tokens/VTokens/VToken.sol

```diff
@@ -346,6 +340,102 @@ contract VToken is VTokenInterface, Exponential, TokenErrorReporter {
+     *      asset by calling the `doTransferOut` internal function.
+     * @param to The address to which the underlying asset is to be transferred.
+     * @param amount The amount of the underlying asset to transfer.
+     * requirements
+     *      - The caller must be the Comptroller contract.
+     * custom:reverts
+     *      - Reverts with "Only Comptroller" if the caller is not the Comptroller.
+     * custom:event Emits FlashLoanAmountTransferred event on successful transfer of amount to receiver
+     */
+    function transferUnderlying(
+        address payable to,
+        uint256 amount
+    ) external nonReentrant returns (uint256 balanceBefore) {
+        if (msg.sender != address(comptroller)) {
+            revert("Invalid comptroller");
+        }
+
+        doTransferOut(to, amount);
+
+        balanceBefore = getCashPrior();
+        emit FlashLoanAmountTransferred(underlying, to, amount);
+    }
+
+    /**
+     * @notice Executes a flashLoan operation.
+     * @dev Transfers the amount to the receiver contract and ensures that the total repayment (amount + fee)
+     *      is returned by the receiver contract after the operation. The function performs checks to ensure the validity
+     *      of parameters, that flashLoan is enabled for the given asset, and that the total repayment is sufficient.
+     *      Reverts on invalid parameters, disabled flashLoans, or insufficient repayment.
+     * @param receiver The address of the contract that will receive the flashLoan and execute the operation.
+     * @param amount The amount of asset to be loaned.
+     * custom:requirements
+     *      - The `receiver` address must not be the zero address.
+     *      - FlashLoans must be enabled for the asset.
+     *      - The `receiver` contract must repay the loan with the appropriate fee.
+     * custom:reverts
+     *      - Reverts with `FlashLoan not enabled` if flashLoans are disabled for any of the requested assets.
+     *      - Reverts with `Execute flashLoan failed` if the receiver contract fails to execute the operation.
+     *      - Reverts with `Insufficient reypayment balance` if the repayment (amount + fee) is insufficient after the operation.
+     * custom:event Emits FlashLoanExecuted event on success
+     */
+    function executeFlashLoan(address payable receiver, uint256 amount) external nonReentrant returns (uint256) {
+        uint256 repaymentAmount;
+        uint256 fee;
+        (fee, repaymentAmount) = calculateFee(receiver, amount);
+
+        IFlashLoanSimpleReceiver receiverContract = IFlashLoanSimpleReceiver(receiver);
+
+        // Transfer the underlying asset to the receiver.
+        doTransferOut(receiver, amount);
+
+        uint256 balanceBefore = getCashPrior();
+
+        // Call the execute operation on receiver contract
+        if (!receiverContract.executeOperation(underlying, amount, fee, msg.sender, "")) {
```

Comment:
What about `params` (the last parameter)?

Reply:
Added.